Templates/MonoGame.Templates.CSharp/content/MonoGame.2D.StartKit/___SafeGameName___.Core/Screens/GameplayScreen.cs

```diff
@@ -0,0 +1,396 @@
+using GameStateManagement.Inputs;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Content;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using Microsoft.Xna.Framework.Input.Touch;
+using Microsoft.Xna.Framework.Media;
+using System;
+using System.IO;
+#endregion
+
+namespace ___SafeGameName___.Screens;
+
+/// <summary>
+/// This screen implements the actual game logic. It is just a
+/// placeholder to get the idea across: you'll probably want to
+/// put some more interesting gameplay in here!
+/// </summary>
+class GameplayScreen : GameScreen
+{
+    #region Fields
+
+    ContentManager content;
+
+    float pauseAlpha;
+
+    private SpriteBatch spriteBatch;
+    Vector2 baseScreenSize = new Vector2(800, 480);
+    private Matrix globalTransformation;
+    int backbufferWidth, backbufferHeight;
+
+    // Global content.
+    private SpriteFont hudFont;
+
+    private Texture2D winOverlay;
+    private Texture2D loseOverlay;
+    private Texture2D diedOverlay;
+
+    // Meta-level game state.
+    private int levelIndex = 0;
+    private Level level;
+    private bool wasContinuePressed;
+
+    // When the time remaining is less than the warning time, it blinks on the hud
+    private static readonly TimeSpan WarningTime = TimeSpan.FromSeconds(30);
+
+    // We store our input states so that we only poll once per frame, 
+    // then we use the same input state wherever needed
+    private GamePadState gamePadState;
+    private KeyboardState keyboardState;
+    private TouchCollection touchState;
+    private AccelerometerState accelerometerState;
+
+    private VirtualGamePad virtualGamePad;
+    private Texture2D backpack;
+
+    // The number of levels in the Levels directory of our content. We assume that
+    // levels in our content are 0-based and that all numbers under this constant
+    // have a level file present. This allows us to not need to check for the file
+    // or handle exceptions, both of which can add unnecessary time to level loading.
+    private const int numberOfLevels = 3;
+    private const int textEdgeSpacing = 10;
+
+    #endregion
+
+    #region Initialization
+
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    public GameplayScreen()
+    {
+        TransitionOnTime = TimeSpan.FromSeconds(1.5);
+        TransitionOffTime = TimeSpan.FromSeconds(0.5);
+    }
+
+    /// <summary>
+    /// LoadContent will be called once per game and is the place to load
+    /// all of your content for the game.
+    /// </summary>
+    public override void LoadContent()
+    {
+        if (content == null)
+            content = new ContentManager(ScreenManager.Game.Services, "Content");
+
+        spriteBatch = ScreenManager.SpriteBatch;
+
+        // Load fonts
+        hudFont = content.Load<SpriteFont>("Fonts/Hud");
+
+        // Load overlay textures
+        winOverlay = content.Load<Texture2D>("Overlays/you_win");
```

Comment:
Should these be using the `LoadLocalized` api? If you are going to have localization you need to make sure you have localized text even if graphics. See https://docs.monogame.net/articles/getting_started/content_pipeline/localization.html#other-localized-assets for details.

Same for fonts, you might be using a different SpriteFont if you are supporting CKJ languages.

Reply:
@dellis1972 Nice pick-up. I'd forgotten about these. I have an idea on how that might work with graphics.

CKJ??

Reply:
Chinese, Korean, Japanese (CKJ). They require specialised fonts which support the character sets they need.